liquidctl/driver/lianli_uni.py

```diff
@@ -0,0 +1,266 @@
+        self.supports_cooling = True
+
+    def initialize(self, **kwargs):
+        """Initialize the device and enable PWM synchronization on all channels."""
+
+        # Enable PWM synchronization on all channels
+        for channel in range(_MIN_CHANNEL, _MAX_CHANNEL + 1):
+            self.toggle_pwm_sync(channel, desired_state=True)
+            time.sleep(0.2)  # Delay to prevent race conditions
+
+        return None
+
+    def get_status(self, **kwargs):
+        """Returns queried speed from controller,
+        if nothing is returned, returns 0"""
+        status = []
+
+        for channel in range(_MIN_CHANNEL, _MAX_CHANNEL + 1):
+            current_speed = self.query_current_speed(channel)
+            duty_name = f"Channel {channel + 1}"
+            if current_speed is None:
+                current_speed = 0.0
+            status.append((duty_name, float(current_speed), "rpm"))
+
+        return status
+
+    def toggle_pwm_sync(self, channel, desired_state=None):
+        """Toggles or explicitly sets PWM synchronization for manual speed control.
+
+        Parameters:
+            channel: int - The zero-based index of the channel
+            desired_state: bool, optional - Set True to enable PWM, False to disable PWM.
+                                            If None, toggle the current state.
+        """
+        if not _MIN_CHANNEL <= channel <= _MAX_CHANNEL:
+            raise ValueError(
+                f"channel must be between {_MIN_CHANNEL} and {_MAX_CHANNEL} (zero-based index)"
+            )
+
+        # Determine the desired action
+        if desired_state is None:
+            # Toggle the current state
+            desired_state = not self.pwm_channels[channel]
+
+        if desired_state:
+            debug_string = "Enabling"
+            channel_byte = 0x11 << channel  # enables PWM
+        else:
+            debug_string = "Disabling"
+            channel_byte = 0x10 << channel  # disables PWM
+
+        # Construct the command to toggle PWM synchronization
+        command_prefix = _PWM_COMMANDS.get(self.device_type)
+        if not command_prefix:
+            raise NotSupportedByDevice(f"Unsupported device type for PWM sync: {self.device_type}")
+
+        command = command_prefix + [channel_byte]
+        _LOGGER.debug("%s PWM sync for channel %d: command %s", debug_string, channel, command)
+
+        try:
+            self.device.write(command)
+            self.pwm_channels[channel] = desired_state  # Update state
+        except Exception as e:
+            _LOGGER.warning("Error writing to device: %s", e)
+
+    def set_fixed_speed(self, channel, duty, **kwargs):
+        """Set a fixed speed for the specified channel.
+
+        Parameters:
+            channel: str or int - The name of the channel (e.g., 'channel1') or the zero-based index of the channel
+            duty: int or float - The desired speed percentage (0-100)
```

Comment:
Be explicit that the additional promises over the base class -- that channel may be an int and that duty may be a float -- are _unstable._

Reply:
duty was changed to int only, however I still force check if its a str as the CMD line passes a str as always. I did update the comment to state that it's the zero-based index only.